servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -330,7 +333,7 @@ public Reference assignReference(
           .canAssignRefToHash(oldRef.getNamedRef())
           .checkAndThrow();
 
-      getStore().assign(oldRef.getNamedRef(), oldRef.getProvidedHash(), newRef.getHash());
+      getStore().assign(oldRef.getNamedRef(), Optional.of(oldRef.getHash()), newRef.getHash());
```

Comment:
I suspect, with this change, the `Optional` in the signatures of `VersionStore.assign` + `VersionStore.create` can go away.

Reply:
Yes. Do you prefer that I make the change now, or in a follow-up PR?

Reply:
Let's remove `Optional` now, because this PR make the hash required and `Optional` unnecessary.

Reply:
Done. We can also make a case for removing the `Optional` in:

* `VersionStore.commit.referenceHash`
* `MergeOp.expectedHash`
* `TransplantOp.expectedHash`

But if everybody agrees, I will look into that in another PR :-)